core/src/main/java/org/elasticsearch/action/DocWriteResponse.java

```diff
@@ -194,7 +200,17 @@ public String getLocation(@Nullable String routing) {
         if (routing != null) {
             location.append(routingStart).append(routing);
         }
-        return location.toString();
+
+        String locationString = null;
+        try {
+            URI uri = new URI(location.toString());
+            locationString = uri.toASCIIString();
+        } catch (URISyntaxException ex) {
```

Comment:
I think I was thinking that it'd be better to just throw the exception and catch it in the caller. That way the caller can log the warning and not add the location header.